### PR TITLE
Support specifying NAS filesystem ID

### PR DESCRIPTION
**Pull request by iltyty**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Support specifying NAS filesystem ID in PV and StorageClass.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/nas/utils.go [hunk 1]
```diff
@@ -307,3 +295,47 @@ func cleanupMountpoint(mounter mountutils.Interface, mountPath string) (err erro
...
+	id, _, _ := strings.Cut(server, "-")
+	return id
+}
+
+func getCPFSIDFromMapOrServer(params map[string]string, server string) string {
+	if id := params[filesystemIDKey]; id != "" {
+		return id
+	}
+	elems := strings.Split(server, "-")
+	return elems[0] + "-" + elems[1]
```
Comment (mowangdk) [inline pkg/nas/utils.go]: Avoid index errors by implementing additional checks.

File: pkg/nas/filesystem_controller_test.go [hunk 2]
```diff
@@ -212,11 +215,12 @@ func TestCreateVolume(t *testing.T) {
...
-					"fileSystemId": "file-system-id",
-					"path":         "/share",
-					"server":       "test.mount.target.domain",
-					"vers":         "3",
+					"deleteVolume":   "false",
+					"fileSystemId":   "file-system-id",
+					"fileSystemType": "extreme",
+					"path":           "/share",
+					"server":         "test.mount.target.domain",
+					"vers":           "3",
```
Comment (mowangdk) [inline pkg/nas/filesystem_controller_test.go:223]: Include test cases for empty fileSystemType and fileSystemId with a custom server.

Comment (iltyty) [reply pkg/nas/filesystem_controller_test.go:223]: I think it's unnecessary since the server field will be ignored when creating new standard nas filesystems.
